src/trace.rs

```diff
@@ -0,0 +1,136 @@
+use crate::event::command::{
+    CommandEventHandler,
+    CommandFailedEvent,
+    CommandStartedEvent,
+    CommandSucceededEvent,
+};
+use bson::Bson;
+
+pub(crate) struct CommandTracingEventEmitter {
```

Comment:
Is the plan to pass the events through this handler after doing tracing if the user also configured an event handler? If so, we'd need something like an `Option<Arc<dyn CommandEventHandler>>` field here.

Reply:
Ah I missed that bit in `emit_command_event`, LGTM. And yeah, creating a new one on the fly seems fine since it doesn't have any state.